habitat-lab/habitat/isaacsim/usd_conversion_scene.py

```diff
@@ -0,0 +1,296 @@
+    convert a scene_instance json to a .usd file by aggregating all the defined 
+    objects within the scene instance json.
+    """
+    
+    # Open scnece_instance JSON
+    with open(scene_instance_filepath, "r") as file:
+        scene_data = json.load(file)
+ 
+    
+    stage_name: str = scene_data[STAGE_INSTANCE][TEMPLATE_NAME].split('/')[1
+    ]
+    
+    # Form a list of existing object glb file paths 
+    object_glb_filepath_list = make_object_filepath_list(
+        glb_data_folder,
+        scene_instance_filepath,
+        scene_data,
+        file_extension=".collider.glb",
+    )
+
+    # # Append stage glb to piggy back on conversion.py. #Note, stage glb not necessary probably
+    #stage_glb_filepath = stage_glb_folder + stage_name + '.glb'
+    #object_glb_filepath_list.append(stage_glb_filepath)
+
+    # Upload glb objects to remote server
+    num_processes = 10
+    barrier = Barrier(num_processes)
+    with Pool(processes=num_processes) as pool:
+        pool.map(upload_object_glb_to_server, object_glb_filepath_list)
+
+    # Process objects remotely
+    run_remote_converter()
+
+    # Download objects
+    download_remote_usd_folder(local_download_folder)
+
+    
+    # TODO: Delete objects on remote server after download
+
+    # Populate USDA file. USDA is the human readable version of a usd file.\
+
+    stage_usd_filepath = local_download_folder + 'object_usd/' + stage_name + '.usd' # NOTE: Might not need this
+    scene_instance_usd_filepath = './' + stage_name + '.scene_instance.usda'
+
+    stage = initialize_stage(stage_usd_filepath, scene_instance_usd_filepath)
+
+    object_usd_folder = local_download_folder + "/object_usd/" #The string is the name of the folder from remote server
+    
+    # This is to ensure the same object used multiple times have a unique ID number
+    previous_object_name = "" 
+    count = 0
+
+    for object_data in scene_data[OBJECT_INSTANCES]:
+        object_exists, usd_filepath, object_name = find_object_usd(
+            object_data, object_usd_folder, file_extension=".collider.usd"
+        )
+
+        # This if/else is to ensure multiple copies of the same type of object have unique names.
+        if previous_object_name == object_name:
+            count += 1
+        else:
+            previous_object_name = object_name
+            count = 0
+
+        if object_exists:
+            add_object_to_stage(stage, object_data, usd_filepath, count)
+
+    stage.GetRootLayer().Save()
+
+    print(f"Finished rendering convertin {scene_instance_filepath} to .usd")
+
+
+if __name__ == "__main__":
+
+    scene_instance_filepath = "/home/trandaniel/dev/habitat-sim/data/hssd-hab/scenes/102343992.scene_instance.json"
+    stage_glb_folder = '/home/trandaniel/dev/IsaacSimConversion/'
+    local_download_folder = "/home/trandaniel/dev/IsaacSimConversion/usd_scene/"
+    glb_data_folder = "/home/trandaniel/dev/habitat-sim/data/hssd-hab/objects"
+
+    convert_scene_instance(
```

Comment:
Let's add `test/test_isaac_sim.py` to habitat-lab in this PR. Look at our other test files in that folder for reference. Let's have a test that tries to convert a toy habitat scene instance (this is something you suggested a while back). The test can assume the source assets are present on the machine (or maybe check for them and skip the test if they're missing). The test should inspect the output `usda` file and check for expected contents.